libraries/app/database_api.cpp

```diff
@@ -362,6 +363,24 @@ optional<block_header> database_api_impl::get_block_header(uint32_t block_num) c
       return *result;
    return {};
 }
+map<uint32_t,block_header> database_api::get_block_header_batch(const vector<uint32_t> block_num)const
+{
+   return my->get_block_header_batch( block_num );
+}
+
+map<uint32_t,block_header> database_api_impl::get_block_header_batch(const vector<uint32_t> block_num) const
+{
+   map<uint32_t,block_header> results;
+   for (const uint32_t& i : block_num) {
+      auto result = _db.fetch_block_by_number(i);
+      if (result)
+         results[i] = *result;
```

Comment:
`results[i] = *result;` is correct because `results` is a map, and the null result is handled in the subsequent line `results[i] = {};`

A minor suggestion for the future: in this case you can prefer `uint32_t` instead of `uint32_t&` in the range-based for loop. Since `uint32 i` is a small native type, it is more efficient to copy the value for each iteration of the loop.